docs/xks/developer-guide/networking.md

```diff
@@ -0,0 +1,320 @@
+outside the cluster.
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-ingress-controller
+spec:
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              name: ingress-nginx
+  podSelector:
+    matchLabels:
+      app: foo
+  policyTypes:
+  - Ingress
+```
+
+#### Allow Cross Namespace
+
+When allowing network traffic across tenant namespaces considerations have to be made for the default deny Network Policy in both namespaces. An allow rule has to be created to allow the source
+namespace, the side initating the connection, to send traffic to the other namespace. The destination namespace has to allow traffic from the source namespace. The first Network Policy should be used
+in the source namespace and the second should be used in the destination namespace.
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-egress-to-destination
+  namespace: source
+spec:
+  egress:
+    - to:
+        - namespaceSelector:
+            matchLabels:
+              name: destination
+          podSelector:
+            matchLabels:
+              app: bar
+  podSelector:
+    matchLabels:
+      app: foo
+  policyTypes:
+    - Egress
+```
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-ingress-from-source
+  namespace: destination
+spec:
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              name: source
+          podSelector:
+            matchLabels:
+              app: foo
+  podSelector:
+    matchLabels:
+      app: bar
+  policyTypes:
+    - Ingress
+```
+
+### Debugging
+
+TBD
+
+## Ingress
+
+[Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) in Kubernetes is used to allow network traffic from the outside the cluster to reach Pods inside the cluster. Ingress works
+as a layer ontop of Kubernetes Services by exposing the Service with a hostname. All Ingress traffic is Layer 7 routed meaning that traffic is routed based on the host header in the HTTP request. This
+also means that Ingress only works with HTTP traffic. Doing it this way means that only a single load balancer is required reducing cost compared to running multiple load balancers, one per Ingress.
+
+<img alt="Ingress Overview" src={useBaseUrl("img/assets/xks/developer-guide/ingress-overview.drawio.jpeg")} />
+
+XKS comes with everything pre configured for Ingress to work. The cluster will either have a single [Nginx Ingress Controller](https://kubernetes.github.io/ingress-nginx/) which is exposed to the
+public internet or two controllers where one is public and one is private. On top of that the cluster is configured with [External DNS](https://github.com/kubernetes-sigs/external-dns) which creates
```

Comment:
Should we mention something about ingressClasses and the needed for it when using multiple ingress?

Reply:
I saw that we have a TBD for private ingress so lets keep that info there.